apps/twap/src/app/utils/computeOrderUid.ts

```diff
@@ -0,0 +1,28 @@
+import type { Order } from '@cowprotocol/contracts';
+import {
+  OrderParameters,
+  OrderSigningUtils,
+  SupportedChainId,
+} from '@cowprotocol/cow-sdk';
+import { returnNever } from '../types/order';
+
+function isOrder(order: Order | OrderParameters): order is Order {
+  return true;
+}
+
+export async function computeOrderUid(
+  chainId: SupportedChainId,
+  owner: string,
+  order: Order | OrderParameters
+): Promise<string> {
+  const { computeOrderUid: _computeOrderUid } = await import(
+    '@cowprotocol/contracts'
+  );
+  const domain = await OrderSigningUtils.getDomain(chainId);
```

Comment:
Actually, you can make a key-value const for chainId-domain

Reply:
Good one, adding a cache for this in proposed way (to handle async nature as well)